runtime/metricsview/executor.go

```diff
@@ -320,3 +322,150 @@ func (e *Executor) Export(ctx context.Context, qry *Query, executionTime *time.T
 		"args": args,
 	})
 }
+
+// SearchQuery executes the provided query against the metrics view.
+func (e *Executor) Search(ctx context.Context, qry *SearchQuery, executionTime *time.Time) ([]SearchResult, error) {
+	if !e.security.CanAccess() {
+		return nil, runtime.ErrForbidden
+	}
+
+	// Generate a metricsview.Query and build a AST
+	// This is a hacky implementation since both metricsview.Query and AST are designed for aggregate queries.
+	// TODO :: Refactor the code and extract common functionality from metricsview.Query and AST and write SearchQuery to underlying SQL/Native druid query directly.
+
+	if e.olap.Dialect() == drivers.DialectDruid {
+		dimensions := make([]Dimension, len(qry.Dimensions))
+		for i, d := range qry.Dimensions {
+			dimensions[i] = Dimension{Name: d}
+		}
+		q := &Query{
+			MetricsView: qry.MetricsView,
+			Dimensions:  dimensions,
+			Where:       qry.Where,
+			Having:      qry.Having,
+			TimeRange:   qry.TimeRange,
+			Limit:       qry.Limit,
+		}
+
+		if err := e.rewriteQueryTimeRanges(ctx, q, executionTime); err != nil {
+			return nil, err
+		}
+
+		ast, err := NewAST(e.metricsView, e.security, q, e.olap.Dialect())
+		if err != nil {
+			return nil, err
+		}
+
+		if err := e.rewriteLimitsIntoSubqueries(ast); err != nil {
+			return nil, err
+		}
+
+		// native search
+		res, err := qry.executeSearchInDruid(ctx, e.olap, ast)
+		if err == nil || !errors.Is(err, errDruidNativeSearchUnimplemented) {
+			return res, err
+		}
```

Comment:
Since this function does execution, I think it might make more sense to put it on the `Executor` type – then you could even move the AST stuff above into the function, and just have something like `e.trySearchInDruid(ctx, qry)` (and you wouldn't need the hack to unquote the table name)